src/js/apps/patients/worklist/worklist_state.js

```diff
@@ -179,18 +182,23 @@ export default Backbone.Model.extend({
     const listName = this.isFlowType() ? 'selectedFlows' : 'selectedActions';
 
     this.set(listName, {});
+    this.set('lastSelectedId', null);
+
     this.trigger('select:none');
   },
-  selectAll(collection) {
+  selectMultiple(collection, newLastSelectedId) {
     const listName = this.isFlowType() ? 'selectedFlows' : 'selectedActions';
 
-    const list = collection.reduce((selected, model) => {
+    const currentSelectedList = this.get(listName);
+
+    const newSelectedList = collection.reduce((selected, model) => {
       selected[model.id] = true;
       return selected;
-    }, {});
+    }, clone(currentSelectedList));
```

Comment:
It doesn't look like `clone(currentSelectedList)` is always `{}` when this function is used by select all. It will contain any items that are currently selected when the select all button is clicked.

I assumed that was okay, since any duplicates are just overridden in that `reduce()` when the new value is set via `selected[model.id] = true;`.

But if there is a better way to do this, I'm open to it 😄

Reply:
Hmm didn't think of that scenario...

Looks like the selected item stays in the `clone(currentSelectedList)` list when it's find-in-list'd out. 

When you bulk edit those items, the changes only apply to the ones visible on the page. Not the non-visible one that still exists in `clone(currentSelectedList)`. So that's good.

But now this makes me nervous too...